src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.LeakDetection/MarkAndCatalogPackages.cs

```diff
@@ -203,15 +174,11 @@ public override bool Execute()
             return !Log.HasLoggedErrors;
         }
 
-        private void Poison(AssemblyDefinition asm)
+        private void Poison(string path)
```

Comment:
Is there a reason `File.AppendAllText` could not be used here?  If so you could declare this as an expression-bodied method

```
private void Poison(string path) => File.AppendAllText(path, PoisonMarker);
```

Reply:
That would probably work fine as well. This was a copy-paste from another repo.

@MichaelSimons if you feel strongly about this and the other comment, I can incorporate them and do offline validation. CI would likely not catch any issues, and I need to ensure everything would still work fine.

Reply:
To me know is the time to fix things like this before they are checked in.  I feel it is reasonable to make these two small changes and just rebuild the tasks locally and not perform a full poison run.  It's your call though.

Reply:
I'm going to do that - thanks!

Reply:
Pushed the new changes.